### DISCUSSION TITLE
Hot-reload tailwind.config

### DISCUSSION THREAD

**Discussion by timfee** [withastro/roadmap#298] [Proposal]
I'm currently doing some hand-tweaking of themes in `tailwind.config.mjs` - and in order to see any changes live, I have to stop/restart my local dev server. Would it be possible to have the tailwind integration support hot reloading when changes to this file occur?

**Comment by natemoo-re** [comment]
IMO we should probably restart the server when any known `*.config.*` files change?
Looking forward to chatting about this on today's [Community Call](https://github.com/withastro/rfcs/discussions/301)!

**Comment by timfee** [comment]
Yes, I noticed that when I manually exit and restart the server, when it
comes back online, the page reloads -- so while that might not be the MOST
optimal/performant solution, it's certainly more than 80% of the way there :)

**Comment by bholmesdev** [comment]
Just discussed at our RFC meeting! This feels trivial enough that we could implement without an RFC. Some notes:
- We agree that reloading for _all_ `.config.*` files in the root of your project (not just Tailwind) would be the most future-proof
- It's worth special casing common base files like `tsconfig.json` and `package.json` (we'll test if Vite respects these changes on restart!)

**Reply by timfee** [reply]
Awesome! And regardless of the fact that the team accepted the RFC, let me just chime in and say how awesome of an open source spirit Astro cultivates! I've had great success interacting with various folks, submitting PRs & the like -- super excited for your continued success.

**Comment by okikio** [comment]
Why not add an option to the `astro.config.mjs` called `watch: { paths: ["..."] }`, where people can state and dictate their configs. I feel us manually doing this for devs may cause unexpected problems down the line, especially due to some configs allowing for extending themselves with other configs
